test/FastTests/Client/Queries/RegexQueryTest.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace FastTests.Client.Queries
+{
+    public class RegexQueryTest : RavenTestBase
+    {
+        [Fact]
+        public void QueriesWithRegexShouldWork()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new RegexMe{Text = "I love dogs and cats"});
+                    session.Store(new RegexMe { Text = "I love cats" });
+                    session.Store(new RegexMe { Text = "I love dogs" });
+                    session.Store(new RegexMe { Text = "I love bats" });
+                    session.Store(new RegexMe { Text = "dogs love me" });
+                    session.Store(new RegexMe { Text = "cats love me" });
+                    session.SaveChanges();
+                    var res = session.Advanced.RawQuery<RegexMe>("from RegexMes as r where Regex(r.Text,\"^[a-z ]{2,4}love\")")
```

Comment:
Create issue to support this from the Linq provider

Reply:
http://issues.hibernatingrhinos.com/issue/RavenDB-9094